src/Interactive/Features/Interactive/Core/InteractiveHost.RemoteService.cs

```diff
@@ -20,7 +20,8 @@ internal sealed class RemoteService
             // output pumping threads (stream output from stdout/stderr of the host process to the output/errorOutput writers)
             private Thread _readOutputThread;           // nulled on dispose
             private Thread _readErrorOutputThread;      // nulled on dispose
-            private InteractiveHost _host;       // nulled on dispose
+            private InteractiveHost _host;              // nulled on dispose
+            private bool _isDisposed;                   // set to true on dispose
```

Comment:
Rename to _disposing to indicate that we started disposing the object, rather than completed the disposal.